3main

```diff
@@ -324,6 +326,60 @@ fi
 
 sleep 2
 
+if [[ $FOREMAN_MONITOR == YES ]] 
+then
+  echo "Running with Foreman integration"
+
+  FOREMAN_HOME=$NVOC/foreman
+
+  source $FOREMAN_HOME/package.sh
+
+  # Verify that the user's client ID and API key are defined
+  if [[ -z $FOREMAN_CLIENT_ID || -z $FOREMAN_API_KEY ]]
+  then
+    echo "Either your FOREMAN_CLIENT_ID or FOREMAN_API_KEY haven't been set - check your 1bash"
+  else
+    FOREMAN_PICKAXE_ID=$(frmn_get_pickaxe_id)
+    if [[ -z $FOREMAN_PICKAXE_ID ]]
+    then
+      # No pickaxe ID - create one for this rig
+      FOREMAN_PICKAXE_ID=$(frmn_create_pickaxe)
+    fi
+
+    # Is a Foreman upgrade needed?
+    if [[ "$(frmn_get_current_pickaxe_version)" != "$(frmn_get_latest_pickaxe_version)" || "$(frmn_get_current_chisel_version)" != "$(frmn_get_latest_chisel_version)" ]]
+    then
+      # Upgrade needed
+      echo "Upgrading Foreman to latest version"
+
+      # Download and run the install/upgrade script
+      wget https://gist.githubusercontent.com/delawr0190/8cb40ab12c9f784d88b80d105e69933b/raw/install.sh \
+        -o /dev/null \
+        -q \
+        -O $FOREMAN_HOME/foreman-install
+      chmod +x $FOREMAN_HOME/foreman-install
+      export FOREMAN_HOME
+      sudo -E bash $FOREMAN_HOME/foreman-install pickaxe $FOREMAN_CLIENT_ID $FOREMAN_API_KEY $FOREMAN_PICKAXE_ID
+      sudo -E bash $FOREMAN_HOME/foreman-install chisel
+
+      # Script is no longer needed
+      rm $FOREMAN_HOME/foreman-install
+    fi
+
+    # Start everything
+    $FOREMAN_HOME/chisel/bin/chisel.sh restart
+    $FOREMAN_HOME/pickaxe/bin/pickaxe.sh restart
+    $FOREMAN_HOME/miner-watch >> $NVOC/nvoc_logs/foreman-miner-monitor &
+  fi
+elif ps ax | grep -q '[f]oreman'
+then
+  # Foreman is no longer being used - stop the running apps
+  $FOREMAN_HOME/chisel/bin/chisel.sh stop
+  $FOREMAN_HOME/pickaxe/bin/pickaxe.sh stop
+fi
+
+sleep 2
```

Comment:
I don't know what this sleep is intended but if you need to pause after starting pickaxe probably this should be moved inside the if block, such that it does not add useless delay in case foreman is disabled.

Reply:
Saw there was a sleep 2 before the Foreman block - I wasn't sure if that was a convention you guys used throughout 3 main, so I put one after.

It's not needed.  I will remove it.